python/ray/experimental/async_plasma.py

```diff
@@ -63,8 +58,14 @@ def as_future(self, object_id, check_ready=True):
 
         future = PlasmaObjectFuture(loop=self._loop)
         self._waiting_dict[object_id].append(future)
-        if not self.check_immediately(object_id) and len(
```

Comment:
This was the bug (`self.check_immediately(object_id)` always returned `None` even if the future was popped so then the following lookup to `self._waiting_dict` would reset the default value to `[]`).